pkg/cmd/roachtest/tests/mixed_version_change_replicas.go

```diff
@@ -253,18 +279,82 @@ func runChangeReplicasMixedVersion(ctx context.Context, t test.Test, c cluster.C
 	}
 
 	// Set up and run test.
-	mvt := mixedversion.NewTest(ctx, t, t.L(), c, c.All(), mixedversion.ClusterSettingOption(
-		// Speed up the queues.
-		install.EnvOption{"COCKROACH_SCAN_MAX_IDLE_TIME=10ms"}),
+	mvt := mixedversion.NewTest(
+		ctx, t, t.L(), c, c.All(), mixedversion.ClusterSettingOption(
+			// Speed up the queues.
+			install.EnvOption{"COCKROACH_SCAN_MAX_IDLE_TIME=10ms"},
+		),
 		// Avoid repeatedly running into #114549 on earlier minor versions.
 		// TODO(kvoli): Remove in 24.2.
 		mixedversion.AlwaysUseLatestPredecessors,
-		// Multi-tenant deployments are currently unsupported. See #127378.
-		mixedversion.EnabledDeploymentModes(mixedversion.SystemOnlyDeployment),
 	)
 
 	mvt.OnStartup("create test table", createTable)
 	mvt.InMixedVersion("move replicas", moveReplicas)
 	mvt.AfterUpgradeFinalized("move replicas", moveReplicas)
 	mvt.Run()
 }
+
+// enableTenantSplitScatter updates cluster settings that allow
+// tenants to use SPLIT AT and SCATTER. This is only performed if the
+// mixedversion test is running on a multitenant deployment, and only
+// if required by the active version.
+func enableTenantSplitScatter(l *logger.Logger, r *rand.Rand, h *mixedversion.Helper) error {
+	settings := []string{
+		"sql.split_at.allow_for_secondary_tenant.enabled",
+		"sql.scatter.allow_for_secondary_tenant.enabled",
+	}
+
+	for _, s := range settings {
+		// Only enable the relevant settings if they are not already
+		// enabled by default.
+		if !h.Context().FromVersion.AtLeast(mixedversion.TenantsAndSystemAlignedSettingsVersion) {
```

Comment:
Nit: Seems like this could be at the top of the function (early return) as it's not specifically tied to `s`.

Reply:
That's true, fixed.